internal/api/jsonrpc.go

```diff
@@ -316,16 +318,29 @@ func (api *API) sendTx(req *acmeapi.APIRequestRaw, payload []byte) *acmeapi.APID
 		ret.Data = &msg
 		return ret
 	}
-	resp, err := api.query.BroadcastTx(genTx)
+	txInfo, err := api.query.BroadcastTx(genTx)
 	if err != nil {
 		msg = []byte(fmt.Sprintf("{\"error\":\"%v\"}", err))
 		ret.Data = &msg
 		return ret
 	}
 
-	api.query.BatchSend()
+	stat := api.query.BatchSend()
+	resp := <-stat
+	close(stat)
```

Comment:
Generally channels should be closed by the sender, not the receiver. As long as the send goes through before the channel is closed, the value will get passed correctly. You can [create a buffered channel](https://play.golang.org/p/OXyqioOzDpb) or you can [block the sender](https://play.golang.org/p/ZheM_s_qoIA).